src/OnboardingSPA/components/Sidebar/components/Customize/DesignFontsPanel/index.js

```diff
@@ -268,12 +268,29 @@ const DesignFontsPanel = forwardRef(
 			( style ) => style.font_content
 		);
 
+		useEffect( () => {
+			const slug = currentData.sitegen?.homepages?.active?.slug;
+			if ( slug && ! customFont.headings ) {
+				const storedCustomFonts =
+					currentData.sitegen.homepages.data[ slug ].customFont;
+				if ( storedCustomFonts ) {
+					setCustomFont( storedCustomFonts );
+				}
+			}
+			// eslint-disable-next-line react-hooks/exhaustive-deps
+		}, [ currentData ] );
+
 		const handleUpdatePreviewSettings = () => {
 			let headings;
 			let body;
 			if ( selectedGroup === 'custom' ) {
 				headings = `var(--wp--preset--font-family--${ customFont.headings })`;
 				body = `var(--wp--preset--font-family--${ customFont.body })`;
+				const slug = currentData.sitegen?.homepages?.active?.slug;
+				if ( slug ) {
+					currentData.sitegen.homepages.data[ slug ].customFont =
```

Comment:
Same case for fonts as well, let's use the `slug` as custom, stored with the data, instead of introducing a new key like `customFont`? We should also try to combine default and custom states to reduce the number of re-renders of the component.